llvm/lib/Transforms/InstCombine/InstCombineSelect.cpp

```diff
@@ -4018,5 +4045,32 @@ Instruction *InstCombinerImpl::visitSelectInst(SelectInst &SI) {
   if (CondVal->getType() == SI.getType() && isKnownInversion(FalseVal, TrueVal))
     return BinaryOperator::CreateXor(CondVal, FalseVal);
 
+  if (SelType->isIntOrIntVectorTy()) {
```

Comment:
```suggestion
  if (SelType->isIntOrIntVectorTy() && (!isa<Constant>(TrueVal) || !isa<Constant>(FalseVal))) {
```
We can early exit if both of select arms are constants.